src/logplex_tlssyslog_drain.erl

```diff
@@ -404,6 +405,23 @@ do_reconnect(State = #state{sock = undefined,
                     %% first failure.
                     ok;
                 _ ->
+                    % inject log message
+                    Msg =  iolist_to_binary(
+                               logplex_syslog_utils:to_msg(
+                                 logplex_syslog_utils:fmt(local7,
+                                                          warning,
+                                                          now,
+                                                          "heroku", 
+                                                          "logplex",
+                                                          "Error L14 (certificate validation): "
+                                                          "error=\"~s" uri=\"~s\"",
+                                                          [error_to_human(Reason), drain_uri(State)]),
+                                 State#state.drain_tok)),
+                    logplex_message:process_msg(Msg,
+                                                State#state.channel_id,
+                                                <<"heroku">>,
+                                                <<"heroku">>,
+                                                logplex_message:shard_info()),
```

Comment:
This seems silly to now export `logplex_message:shard_info/0` when the call is not being avoided ever. Why just always call shard_info() in process_msg? Or is there some magic incantation of process_msg that gets called with *different* ShardInfo and the process_msg/5 is really necessary?